content/cli/workspaces.md

```diff
@@ -0,0 +1,206 @@
+
+It's important to note that virtually all of Nest's features are independent of your code organization mode. The **only** affect of this choice is how your projects are composed and how build artifacts are generated.  All other functionality, from the CLI to core modules to add-on modules work the same in either mode.
+
+Also, you can easily switch from **single project mode** to **workspace mode** at any time, so you can delay this decision until the benefits of one or the other approach become more clear.
+
+### Single project mode
+
+When you run `nest new`, a new **project** is created for you using a built-in schematic. The basic steps are:
+
+1. Create a new folder, corresponding to the `name` option you provide to `nest new`
+2. Populate that folder with default files corresponding to a minimal base-level Nest application.  You can examine these files at the [typescript-starter](https://github.com/nestjs/typescript-starter) repository.
+3. Provide additional files such as `nest-cli.json`, `package.json` and `tsconfig.json` that configure and enable various tools for compiling, testing and serving your application.
+
+From there, you can modify the starter files, add new components, add dependencies (e.g., `npm install`), and otherwise develop your application as covered in the rest of this documentation.
+
+### Workspace mode
+
+To enable workspace mode, you start with a *single project mode* structure, and add **projects**.  A project can be a full **application** (which you add to the workspace with the command `nest generate app`) or a **library** (which you add to the workspace with the command `nest generate library`). We'll discuss the details of these specific types of project components below.  The key point to note now is that it is the **act of adding a project** to an existing single project mode structure that **converts it** to workspace mode.  Let's look at an example.
+
+If we run:
+```bash
+nest new my-project
+```
+
+We've constructed a *single project mode* structure, with a folder structure that looks like this:
+
+<div class="file-tree">
+  <div class="item">src</div>
+  <div class="children">
+    <div class="item">app.controller.ts</div>
+    <div class="item">app.service.ts</div>
+    <div class="item">app.module.ts</div>
+    <div class="item">main.ts</div>
+  </div>
+  <div class="item">node_modules</div>
+  <div class="item">nest-cli.json</div>
+  <div class="item">package.json</div>
+  <div class="item">tsconfig.json</div>
+  <div class="item">tslint.json</div>
+</div>
+
+The `nest-cli.json` file looks like this:
+```json
+{
+  "collection": "@nestjs/schematics",
+  "sourceRoot": "src"
+}
+```
+
+We can convert this to a *workspace mode* structure as follows:
```

Comment:
We should mention here that this won't work for apps that didn't follow our recommendations regarding the directories structure etc. By default, we try to move `src` and `test` folder. If one changed the original name, that would be a problem.